pkg/autodiscovery/autoconfig.go

```diff
@@ -190,6 +193,16 @@ func (ac *AutoConfig) AddConfigProvider(provider providers.ConfigProvider, shoul
 func (ac *AutoConfig) LoadAndRun() {
 	resolvedConfigs := ac.GetAllConfigs()
 	ac.schedule(resolvedConfigs)
+	atomic.StoreUint32(&ac.ranOnce, 1)
+	log.Debug("LoadAndRun done.")
+}
+
+// HasRanOnce returns true if the AutoConfig has ran once.
+func (ac *AutoConfig) HasRanOnce() bool {
+	if ac == nil {
+		return false
+	}
```

Comment:
I think this check is useless? (if it's nil we'd get a nil pointer dereference when calling this)

Reply:
It is useful in the case of somebody calling `HasRunOnce` too soon on a nil instance of the AutConfig, because no we can call a method on a nil instance of something, this would not crash. Let's discuss this offline.

Reply:
you're right, I stand corrected, thanks for clarifying 👍